### PR TITLE
Add image module

### PR DESCRIPTION
**Pull request by ovekyc**
- Edit clearDB function
- Edit apidoc rule (include image module)
- Add Image controller
- Add image controller function get
- Add test code image controller get
- Add Image manager
- Add fetchPrefix method at ImageManager
- Add test code for fetchPrefix of ImageManager
- Add Apidoc for image-get
- Add getPrefix function at KeyUtils

### CODE REVIEW

**Comment by ovekyc** [comment]
- rebase
- fix lint warning

File: src/server/database.js [hunk 1]
```diff
@@ -9,9 +9,9 @@ export const clearDB = (cb) => {
   db.close(() => {
     leveldown.destroy(config.database, (err) => {
       if (err) {
-        // error
+        return cb(err);
       }
-      db.open(cb);
+      return db.open(cb);
```
Comment (hewonjeong) [inline src/server/database.js]: good!

File: src/server/images/controllers.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import {ENTITY, STATE, KeyUtils} from '../key-utils';
+import {S3Connector} from '../aws-s3';
+import {APIError} from '../ErrorHandler';
+import ImageManager from './models';
+
+export default {
+  get(req, res, cb) {
+    const {itemid, imageid} = req.query;
+    if (itemid) {
```
Comment (festinalente91) [inline src/server/images/controllers.js]: why don't you use Camelcase?

File: src/server/images/controllers.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        }));
+      }
+      Promise.all(promises).then(() => {
+        const conn = new S3Connector();
+        conn.getImageUrls(keys, (err, urls) => {
+          if (err) {
+            return cb(new APIError(err));
+          }
+          res.status(200).send({
+            imageUrls: urls
```
Comment (festinalente91) [inline src/server/images/controllers.js]: How about just using the variable name as 'imageUrls'?

File: src/server/images/models.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import db from '../database';
+
+export default class ImageManager {
+  static fetchPrefix(prefix, cb) {
```
Comment (festinalente91) [inline src/server/images/models.js:4]: why did you use static?

File: src/server/items/controllers.js [hunk 5]
```diff
@@ -173,7 +173,7 @@ export default {
       endTime: req.body.endTime,
       createdDate: currentTime.toISOString(),
       modifiedDate: currentTime.toISOString()
-    }
+    };
```
Comment (hewonjeong) [inline src/server/items/controllers.js:201]: thx...☆

File: src/server/images/controllers.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+          const prefix = KeyUtils.getPrefix(ENTITY.IMAGE, state, itemid);
+          ImageManager.fetchPrefix(prefix, (err, data) => {
+            if (err) {
+              reject(err);
+              return;
+            }
+            data.map((key) => {
+              keys.push(key);
+            });
+            resolve();
```
Comment (hewonjeong) [inline src/server/images/controllers.js:26]: We get origin image's keys in data arr using this fetchPrefix function.
We have to copy element's of data array to keys array?
I think we can just call 'resolve(data)' to send keys array to .then(keys) block.

File: src/server/images/models.js (hunk 4 above)
Comment (ovekyc) [reply src/server/images/models.js:4]: static make to use method without creating an instance.
I just want to use this function without create instance That is it!

File: src/server/database.js (hunk 1 above)
Comment (festinalente91) [reply src/server/database.js]: sorry I changed this code to promise based code

**Comment by HewonJeong** [comment]
LGTM

**Comment by festinalente91** [comment]
LGTM

**Comment by hewonjeong** [comment]
LGTM

**Pull request merged**